config/peerpods/podvm/libvirt-podvm-image-handler.sh

```diff
@@ -21,23 +21,75 @@ function verify_vars() {
 
     [[ -z "${ORG_ID}" ]] && error_exit "ORG_ID is not set"
     [[ -z "${ACTIVATION_KEY}" ]] && error_exit "ACTIVATION_KEY is not set"
-    [[ -z "${BASE_OS_VERSION}" ]] && error_exit "BASE_OS_VERSION is not set"
 
-    [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
+    if [[ "${IMAGE_TYPE}" == "operator-built" ]]; then
+        [[ -z "${BASE_OS_VERSION}" ]] && error_exit "BASE_OS_VERSION is not set"
 
-    [[ -z "${CAA_SRC}" ]] && error_exit "CAA_SRC is empty"
-    [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
+        [[ -z "${PODVM_DISTRO}" ]] && error_exit "PODVM_DISTRO is not set"
 
-    [[ -z "${REDHAT_OFFLINE_TOKEN}" ]] && error_exit "Redhat token is not set"
+        [[ -z "${CAA_SRC}" ]] && error_exit "CAA_SRC is empty"
+        [[ -z "${CAA_REF}" ]] && error_exit "CAA_REF is empty"
 
-    # Ensure booleans are set
-    [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+        [[ -z "${REDHAT_OFFLINE_TOKEN}" ]] && error_exit "Redhat token is not set"
+
+        # Ensure booleans are set
+        [[ -z "${DOWNLOAD_SOURCES}" ]] && error_exit "DOWNLOAD_SOURCES is empty"
+    fi
 }
 
+# Function to create the libvirt image based on the type.
+function create_libvirt_image() {
+    # If the value of the PODVM_IMAGE_URI is empty, then build the image from scratch else use the prebuilt artifact.
+    if [[ "${IMAGE_TYPE}" == "operator-built" ]]; then
+        create_libvirt_image_from_scratch
+    elif [[ "${IMAGE_TYPE}" == "pre-built" ]]; then
+        create_libvirt_image_from_prebuilt_artifact
+    fi
 
+}
 
-function create_libvirt_image() {
-    echo "Creating libvirt qcow2 image"
+# Function to create the libvirt image from a prebuilt artifact.
+function create_libvirt_image_from_prebuilt_artifact() {
+    echo "Pulling qcow2 image for libvirt provider from the provided path"
+    PODVM_IMAGE_SRC="/tmp/image"
+    PODVM_DESTINATION_PATH="/podvm"
+    PODVM_ROOTFS_PATH="/rootfs"
+    PODVM_IMAGE_REPO_AUTH_FILE="/tmp/regauth/auth.json"
+
+    get_image_type_url_and_path
+
+    case "${PODVM_IMAGE_TYPE}" in
+    oci)
+        echo "Extracting the Libvirt qcow2 image from the given path."
+
+        mkdir -p "${PODVM_DESTINATION_PATH}" ||
+            error_exit "Failed to create the pause_bundle directory"
+        
+        extract_container_image $PODVM_IMAGE_URL $PODVM_IMAGE_TAG $PODVM_IMAGE_SRC $PODVM_DESTINATION_PATH $PODVM_IMAGE_REPO_AUTH_FILE
+
+        # Form the path of the podvm qcow2 image.
+        PODVM_IMAGE_PATH="${PODVM_DESTINATION_PATH}${PODVM_ROOTFS_PATH}${PODVM_IMAGE_SRC_PATH}"
```

Comment:
/podvm/rootfs/tmp/image/image/podvm.qcow2 ?
Is this right, /rootfs is fixed right?

A general comment, i think it would be better to avoid defining global variables but simply pass it to the the function using it, if you do need global variables, you can do the defining  and exporting in one line (not sure export is required though)

Reply:
the `PODVM_IMAGE_PATH` will be `/podvm/rootfs/image/podvm.qcow2`.

`PODVM_DESTINATION_PATH` is `/podvm` which is the folder created using `mkdir` before invoking `extract_container_image`. `PODVM_ROOTFS_PATH` is always `/rootfs` as you correctly pointed and then the `PODVM_IMAGE_SRC_PATH` could be `/image/podvm.qcow2` or the one which customer gives as `<image_path>`, so it formulates to `/podvm/rootfs/image/podvm.qcow2`.

let me remove the variables and pass it directly as you suggested..

Reply:
`IMAGE_SRC` which is `/tmp/image` is where the `skopeo` copy is made to using,
```
$SKOPEO_CLI "docker://${docker_image_repo_url}:${image_tag}" "oci:${image_src}:${image_tag}" 
```
and then further from here the `umoci` command puts it in the `PODVM_DESTINATION_PATH`.